lib/datadog/core/environment/identity.rb

```diff
@@ -49,9 +49,18 @@ def lang_version
           Core::Environment::Ext::LANG_VERSION
         end
 
+        # Returns tracer version, rubygems-style
         def tracer_version
           Core::Environment::Ext::TRACER_VERSION
         end
+
+        # Returns tracer version, comforming to https://semver.org/spec/v2.0.0.html
+        def tracer_version_semver2
+          # from ddtrace/version.rb, we have MAJOR.MINOR.PATCH plus an optional .PRE
+          # - transform .PRE to -PRE if present
+          # - keep triplet before that
+          tracer_version.sub(/\.([a-zA-Z0-9]*[a-zA-Z][a-zA-Z0-9]*.*)$/, '-\1')
```

Comment:
I find the regex a little grok (Probably because I never took the time to master it)

Wondering if having a hardcoded constant in `version.rb` would do the trick:

```ruby
  STRING = [MAJOR, MINOR, PATCH, PRE].compact.join('.')
  SEMVER2 = PRE ? "#{MAJOR}.#{MINOR}.#{PATCH}-#{PRE}" : STRING
```

Reply:
I find this pattern more legible as well.